x/core/keeper/genesis.go

```diff
@@ -8,7 +8,25 @@ import (
 
 // InitGenesis initializes the store based on the given genesis state.
 func (k Keeper) InitGenesis(ctx sdk.Context, data types.GenesisState) {
-	err := k.SetParams(ctx, data.Params)
+	// set owner state
+	err := k.SetOwner(ctx, data.Owner)
+	if err != nil {
+		panic(err)
+	}
+
+	// set pending owner state
+	err = k.ClearPendingOwner(ctx)
```

Comment:
I think this should be `SetPendingOwner(ctx, data.PendingOwner)`?

Reply:
Would get rejected since empty string is not a valid address :x

Reply:
I'll move that check to the tx function and out from the setter